Settlement/Total.ts

```diff
@@ -14,6 +15,16 @@ export namespace Total {
 	export function add(addendee: Total, addend: Total): Total {
 		return { amount: Amounts.add(addendee.amount, addend.amount), fee: Fee.add(addendee.fee, addend.fee) }
 	}
+	export function compare(expected: Total, outcome: Total): boolean {
+		return (
+			([...new Set([...Object.keys(expected.amount), ...Object.keys(outcome.amount)])] as isoly.Currency[]).every(
+				currency => expected.amount[currency] == outcome.amount[currency]
+			) &&
+			([...new Set([...Object.keys(expected.fee.other), ...Object.keys(outcome.fee.other)])] as isoly.Currency[]).every(
+				currency => expected.fee.other[currency] == outcome.fee.other[currency]
+			)
+		)
+	}
```

Comment:
It's simpler to use Object.entries(expected.amount).every(([currency, amount] => outcome.amount[currency] == amount))

Reply:
Also a function which is typically good to have a test for. (You don't have to but would be nice.

Reply:
I'm trying to cover for when "expected" might have fewer currencies than "outcome", but maybe that requires some different error handling?

Reply:
Fair enough.